### ISSUE TITLE
[feature] to create new tab for word in main window for "Send word to main window"

### ISSUE THREAD

**Issue by Aliaksandr3by** [goldendict/goldendict#1353]
by click on "Send word to main window", the preprevious tab  is rewriting, because of this need to click  "back";
i think, if the tab is not "untited", program may create new tab for word;

![image](https://user-images.githubusercontent.com/11432724/108286955-14eedc80-719b-11eb-9415-8a274c4e1d7b.png)

**Comment by vedgy** [comment]
On the other hand, if the user sends many words to the main window, it can end up with too many unwanted tabs if a new one is created for each sent word. For most users this would only require extra work of closing them. History navigation is easy with the history pane and with the back/forward buttons. If you want to keep your tab intact, create a new one and let it be overwritten.

**Comment by aksr** [comment]
I think this is a good idea.
@vedgy: I'm not sure I follow you.., the sent  word could be treated as a string?

**Comment by vedgy** [comment]
@aksr, I am sure I don't follow *you* :)
What do you mean by "treating the sent word as a string"?

My point above is that creating a new tab for each sent word results in an unbounded number of tabs. The user has an extra job to close them, which is not straightforward if (s)he has more than one valuable tab that shouldn't be closed (i.e. when *Close all tabs except current* can't be used). The large number of open tabs can waste lots of memory if they are not closed soon.

**Comment by Aliaksandr3by** [comment]
![image](https://user-images.githubusercontent.com/11432724/108360934-e60c5100-7202-11eb-8a0b-9ec9686e437d.png)

to give a chose;

![image](https://user-images.githubusercontent.com/11432724/108361243-41d6da00-7203-11eb-9df9-7e7034f51c66.png)

i think, close tab best way, then to use history.

**Comment by vedgy** [comment]
I don't think this option is worth spending time to implement. But I wouldn't object if someone contributes the option - as long as it is off by default.

**Comment by aksr** [comment]
@vedgy: If I understood correctly, he doesn't mean every word as if in a sentence, for example... It would search for a whole sentence, and every new search opens a *new* tab.

**Comment by vedgy** [comment]
@aksr, all right, I understand you now. What I meant by *sent word* was actually the entire *sent phrase* as per current imprecise GoldenDict's UI text. My point was that a new tab would be created each time the user sends a _phrase_ to the main window, and an average user might not realize that the newly created tabs should be closed to save memory, especially if (s)he doesn't care about any of the loaded articles after reading them.

**Comment by Aliaksandr3by** [comment]
@vedgy : i do not see problem with memory .
might you add new button for sending word in main window as new TAB? and leave old button simultaneously?

and Why Pop Menu don't have a "enter" button for start to finding the word? I like use only mouse :)

![image](https://user-images.githubusercontent.com/11432724/109455046-c4169800-7a66-11eb-9aab-2bc4b117abec.png)

![image](https://user-images.githubusercontent.com/11432724/109454836-3470e980-7a66-11eb-9747-d2275b4e3e40.png)

**Comment by vedgy** [comment]
> @vedgy : i do not see problem with memory .

I've monitored GoldenDict's RAM usage in a system monitor. It grows by megabytes when new tabs are created and shrinks when the tabs are closed.

> might you add new button for sending word in main window as new TAB? and leave old button simultaneously?

I think there are too many buttons in the scan popup already. Ctrl+click on the existing *Send word to main window* button to send in a new tab would be more conventional and less cluttering.

> and Why Pop Menu don't have a "enter" button for start to finding the word? I like use only mouse :)

I don't understand how you can enter a search phrase using only the mouse to begin with. You have to use the keyboard to modify the search line. And then it's way easier to press *Return*/*Enter* than point the mouse pointer at a small button and click. This feature has also been requested and discussed in #1300.

**Comment by Aliaksandr3by** [comment]
Ctrl+click sounds is good)))
p.s. one problem in one time;
i use 99% ctrl+c+c to send word from the PDF, only 1% of word i edit by hand

**Comment by vedgy** [comment]
> i use 99% ctrl+c+c to send word from the PDF, only 1% of word i edit by hand

In that case there is no need for the *translate* button as the article for the word sent by *Ctrl+C+C* is loaded automatically. When you edit a word by hand, it's quite convenient to press *Return*/*Enter* in the end.
